# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchStreamReader.h
# Commit: cc893e08aa2f
# Full Hash: cc893e08aa2f8e98312e33eb5af8f684b015c29c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2022-08-03 03:42:48
# Regressor Bug: 1770630
# File Overlap Count: 1
# Description:
#   Bug 1770630 - Worker stream readers should contribute to busy count. r=dom-worker-reviewers,jstutte
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D149185
# ==============================================================================

diff -r 57951e2dc196 -r cc893e08aa2f dom/fetch/FetchStreamReader.h
--- a/dom/fetch/FetchStreamReader.h	Tue Aug 02 20:55:25 2022 +0000
+++ b/dom/fetch/FetchStreamReader.h	Tue Aug 02 20:56:53 2022 +0000
@@ -19,7 +19,7 @@
 
 class ReadableStream;
 class ReadableStreamDefaultReader;
-class WeakWorkerRef;
+class StrongWorkerRef;
 
 class FetchStreamReader final : public nsIOutputStreamCallback {
  public:
@@ -66,7 +66,8 @@
 
   nsCOMPtr<nsIAsyncOutputStream> mPipeOut;
 
-  RefPtr<WeakWorkerRef> mWorkerRef;
+  RefPtr<StrongWorkerRef> mWorkerRef;
+  RefPtr<StrongWorkerRef> mAsyncWaitWorkerRef;
 
   RefPtr<ReadableStreamDefaultReader> mReader;
 